internal/pkg/api/server/buffer/dependencies/store.go

```diff
@@ -34,21 +36,10 @@ func ProjectKey(projectID int) string {
 	return fmt.Sprintf("config/%d", projectID)
 }
 
-func (c *ConfigStore) CountReceivers(ctx context.Context, projectID int) (count uint64, err error) {
-	logger, tracer, client := c.logger, c.tracer, c.etcdClient
-
-	_, span := tracer.Start(ctx, "kac.api.server.buffer.dependencies.store.CreateReceiver")
-	defer telemetryUtils.EndSpan(span, &err)
-
-	key := ProjectKey(projectID)
-
-	logger.Debugf(`Reading "%s" count`, key)
-	r, err := client.KV.Get(ctx, key, etcd.WithPrefix(), etcd.WithCountOnly())
-	if err != nil {
-		return 0, err
-	}
+type ReceiverLimitReached struct{}
 
-	return uint64(r.Count), nil
+func (*ReceiverLimitReached) Error() string {
+	return "receiver limit reached"
```

Comment:
Tu by bolo dobre vypisat viac info.
`cannot create receiver: there are already "100" receivers in the project, the maximum is "100"`

Takto:
- uzivatel bude vediet aky je limit a nebude ho musel hladat v docs
- ty ako programator z chyby vidis, ze vsetko funguje spravne, ... ked tam bude `already "102" receivers`, tak uz vies, ze je tam aj nejaky problem, zle spraveny zamok ....

Najjednoduchsie je spravit:
```
ReceiverLimitReached struct{
  error
}
```
![image](https://user-images.githubusercontent.com/19371734/202167255-40c32b29-4637-4f7a-a615-d6378ab4b0da.png)
![image](https://user-images.githubusercontent.com/19371734/202167313-ffa6fecb-5a49-4558-86ce-dcf9c9c512aa.png)

ak tu chybu nepotrebujes spracovat nejak v kode:
- ine moznosti su, definovat `message string`
- definovat `actual int` a `maximum int` a hlasku vytvorit v `Error` metode

Reply:
BTW ak chces z tej interneh chyby spravit vetu, a rovno ju vratit cez API, tak mozes pouzit
`errors.Format(err, errors.FormatAsSentences())` (nas `errors` balicek).

Ak nie, tak asi budes musel ist tou cestou:
> definovat actual int a maximum int a hlasku vytvorit v Error metode

Pretoze tie cisla chceme vratit aj z API, a zo `string` spravy ich nedostanes.

Reply:
Lepší chybu zajišťuje ta service (píše tam kolik + přidává status kód). Tahle error message se nikde nezobrazí, je tam ta metoda jen aby to implementovalo interface.

Reply:
Zobrazi sa, v teste, alebo ked ten `store` pouzije worker a pod ...

Reply:
Nikde receiver limit netestujeme, ale pokud se to bude zobrazovat jen interně, tak ta error hláška pro nás stačí

Reply:
Ano tam mas potom to maximum, ale uz nie aktualny pocet (ktory by sa mohol hodit).

Na tom aktualnom pocte netrvam, ak ti nedava zmysel.
Ale to, ze kazda chyba by mala mat dostat informacii (aj ked ju aktualne uzivatel neuvidi) mi pride dolezite.

Vyzera to ako workaround pre to `error.Is` nizsie, lebo z dynamickou message, alebo `actual count` v error, by to neslo.

Reply:
Když jsem tuhle metodu přidal, tak jsem nemyslel na to, že níž s `error.Is` budu muset vytvářet hodnotu a bylo by to složitější s dynamickou message. Jen jsem potřeboval, aby ta vracená hodnota implementovala `error`. Ta chybová hláška má dostatek informací, protože receiver limit se (zatím) nemění.

Reply:
Upraveno, count v erroru ale zatím není, když je to konstanta